engine/src/main/java/io/camunda/zeebe/engine/Engine.java

```diff
@@ -130,9 +132,13 @@ public ProcessingResult process(
         return processingResultBuilder.build();
       }
 
-      final boolean isNotOnBlacklist =
-          !processingState.getBlackListState().isOnBlacklist(typedCommand);
-      if (isNotOnBlacklist) {
+      if (
```

Comment:
Also, instead of dropping comments to the if statement, it would be more clear if we assign first two conditions to a variable (e.g. `noBlacklistCheckNeeded`) and the second one also (e.g `isNotOnBlacklist`).

Reply:
for the second condition `!processingState.getBlackListState().isOnBlacklist(typedCommand))` we want to avoid it's executing completely in cases where a blacklist check does not make sense, thus I would keep it inline.
Happy to extract the first part though.

Reply:
Right, you do not want to evaluate the second one early. 👍